universal-application-tool-0.0.1/app/services/program/ProgramQuestionDefinition.java

```diff
@@ -31,12 +40,31 @@ public boolean hasQuestionDefinition() {
   }
 
   @JsonCreator
-  static ProgramQuestionDefinition create(@JsonProperty("id") long id) {
-    return new AutoValue_ProgramQuestionDefinition(id, Optional.empty());
+  static ProgramQuestionDefinition create(
+      @JsonProperty("id") long id, @JsonProperty("required") boolean required) {
+    return new AutoValue_ProgramQuestionDefinition(id, required, Optional.empty());
   }
 
+  /** Create an optional program question definition. */
   public static ProgramQuestionDefinition create(QuestionDefinition questionDefinition) {
+    return create(questionDefinition, false);
+  }
+
+  /** Create a program question definition. */
+  public static ProgramQuestionDefinition create(
+      QuestionDefinition questionDefinition, boolean required) {
+    return new AutoValue_ProgramQuestionDefinition(
+        questionDefinition.getId(), required, Optional.of(questionDefinition));
+  }
+
+  /** Return a program question definition with the {@link QuestionDefinition} set. */
+  public ProgramQuestionDefinition setQuestionDefinition(QuestionDefinition questionDefinition) {
     return new AutoValue_ProgramQuestionDefinition(
-        questionDefinition.getId(), Optional.of(questionDefinition));
+        questionDefinition.getId(), required(), Optional.of(questionDefinition));
+  }
+
+  /** Return a program question definition with a new required setting. */
+  public ProgramQuestionDefinition setRequired(boolean required) {
```

Comment:
question (non-blocking): Should we use a builder here instead?

Reply:
I'd rather not. It isn't so complex that it needs a builder yet.